Feature/periodogram improvements

jnation3406 opened the pull request:
There are a bunch of changes in here so it might be easiest to just talk it over, but I'll try to at least put an outline here:

- Support for other input types in `OperationWizard` - including new 'source' type with its own `SourceInputWidget` which can do a source lookup by name and provides a name/ra/dec to the operation
- Changes to `MultiImageInputSelector` used by the wizard to allow restricting by filter, and allow selecting all images of a given filter from a dropdown - keyed on new keywords in the operations wizard description
- Refactor `ImageGrid` into a more generic `OperationOutputGrid` that can support both images and non-image data output in a grid view. The `ImageGrid` still exists since it is used for input image selection, which is kind of different now from showing output
- New `DataOutput` which encapsulates generic data output (as opposed to image output) from operations. It has code to show a sparkline plot of light curve or period data as well as the source / operation name
- Rename `AnalysisView` into `ImageAnalysisView` and only launch it from image data.
- Create `DataAnalysisView` which is launched from `DataOutput` data. This is where your code for showing the light curve, periodogram, and phased light curve will go. The more generic and piecemeal this is the better, since we will likely have to extend this to show other data output stuff later with new operations.

Commit:
Some fixes in the current periodogram plot stuff

capetillo committed:
Periodogram changes to store adds logic to store frequency and power. pairs frequency and power and finds peak power and peakindex. stores periodogram. moved foldperiod to be its own function. and applies selected period

capetillo committed:
Adds periodogram chart to analysis view

Periodogram generates a dynamic phased curve based on the point
that the user selects.
Added styling as well.

capetillo committed:
Periodogram Component Added

Component stores values in store and reacts to user's selections.
Currently, not sensitive enough to clicks.

Commit:
More changes to support the variable star operation

capetillo committed:
Merge remote-tracking branch 'origin/feature/periodogram_improvements' into feature/period-and-dpc

Merging Jon's base branch with my changes

capetillo commented on src/components/DataSession/Operation/MultiImageInputSelector.vue:
```diff
@@ -54,14 +55,54 @@ const fitsImages = computed(() => {
+}
+
 // Image dragged into the selected images area
 function insert(inputKey, index, event) {
```
just a suggestion, would it make sense to make this function into a utils function since other operations use `insert` logic?

capetillo reviewed (approved):
LGTM! P.S. Have we figured out why the dragging scale lost its blue color?

capetillo committed:
Updates on store (and more to come)

Removes code related to phased light curve

capetillo committed:
Removes code related to variable stars and phased light curves

capetillo committed:
Passes data through props & adds dynamic phased curve & periodogram

Data was obtained through the store. Now it's through props.
The phase curve is now dynamic. When user clicks on a point in the periodogram
the phase curve updates its values
Adds periodogram logic

capetillo committed:
fixes lint issue

capetillo committed:
Adds logic to view components as data outputs

Data is now passed through props instead of through the store.
Logic to fold period and manage data is also here instead of in store.

capetillo committed:
adds header and styles

capetillo committed:
Adds Logic to Download LightCurve Chart as PNG
The chart downloads as png and all items are black

capetillo committed:
fixes styling

capetillo committed:
more styling and logic to download periodogram as png

capetillo committed:
moves logic to download chart to a utils function

capetillo committed:
uses util function to download chart

capetillo committed:
removes test code

capetillo committed:
reverses y-axis so magnitude is reflected correctly

capetillo committed:
Renames component from PeriodPlot to PhasedLightCurvePlot; lint changes (automatic); and import updates

capetillo committed:
Adds code -- removed all of it by accident

capetillo committed:
removes unused code

capetillo committed:
backend already returns values sorted. removes sorting logic

capetillo committed:
removes unused code

capetillo committed:
Merge pull request #247 from LCOGT/feature/period-and-dpc

Feature/period and dpc
